lldb/source/Commands/CommandObjectHealthcheck.cpp

```diff
@@ -52,15 +52,14 @@ bool CommandObjectHealthcheck::DoExecute(Args &args,
 
   result.AppendMessageWithFormat("Health check written to %s\n",
                                  temp_path.c_str());
-#if defined(__APPLE__)
+
   // When in an interactive graphical session and not, for example,
   // running LLDB running over ssh, open the log file straight away in
   // the user's configured editor or the default Console.app otherwise.
   if ((strcmp("lldb", getprogname()) == 0 ||
```

Comment:
@adrian-prantl this fails on Linux:
```
/home/build-user/llvm-project/lldb/source/Commands/CommandObjectHealthcheck.cpp:59:23: error: use of undeclared identifier 'getprogname'
  if ((strcmp("lldb", getprogname()) == 0 ||
                      ^
/home/build-user/llvm-project/lldb/source/Commands/CommandObjectHealthcheck.cpp:60:34: error: use of undeclared identifier 'getprogname'
       strcmp("lldb-rpc-server", getprogname()) == 0) &&
                                 ^
2 errors generated.
```

Reply:
Right — there's a reason for this. Let's just not take that commit.

Reply:
Okay... This commit was actually a commit between some others and so ended up being skipped (which makes sense). So I need to cut this one out. Though note that it seems to have changed slightly between stable/20211026 and next - next has `__APPLE__` and stable/20211026 has `APPLE`.

And looking at the full history of this file, we're also missing 865688b4c88e9be463abf32b472ed78f973f1a9d. That also isn't on next.

Reply:
I've removed this commit. Up to you if you care about the APPLE vs __APPLE__. The other commit cherries later.